proto/client/gClient.go

```diff
@@ -160,8 +160,9 @@ func (confClient *ConfigClient) subscribeToConfigPod(commChan chan *protos.Netwo
 	logger.GrpcLog.Infoln("subscribeToConfigPod ")
 	myid := os.Getenv("HOSTNAME")
 	var stream protos.ConfigService_NetworkSliceSubscribeClient
+	maxRetryCount := 10
 	for {
-		if stream == nil {
+		if stream == nil && maxRetryCount > 0 {
```

Comment:
What happens if we fail.. I thought unlimited try is better right ? 
What if webconsole does not come up in (5*10 = 50 sec) 50 seconds?

Reply:
Hello,
Here the real problem is if SMF has the wrong IP resolution of NMS (if NMS pod is restarted), the loop never exits and without manually restarting the SMF pod, SMF can not connect to Webui again.
Putting a max retry count causes to restart the SMF pod and it gets the correct GRPC server address. 
No matter GPRC server is up or not. Untill GRPC server is started it this process repeats. 
That helps us to recover 5G configuration synchronization.
Otherwise, we need to detect all modules which have issues to connect NMS and restart them manually.

Reply:
Here are the full logs of NRF: https://pastebin.ubuntu.com/p/4cPQVvVdWv/

NMS is restarted then NRF lost the connection to NMS. After 10 retry, NRF is restarted and GRPC connection is recoverd.

Reply:
Understand what you are trying to solve. Could you point me to the code where SMF would restart after all MaxRetry?

Reply:
In SMF  https://github.com/omec-project/smf/blob/master/factory/factory.go#L52 calls  PublishOnConfigChange function.

```
		roc := os.Getenv("MANAGED_BY_CONFIG_POD")
		if roc == "true" {
			gClient := ConnectToConfigServer(SmfConfig.Configuration.WebuiUri)
			commChannel := gClient.PublishOnConfigChange(false)
			go SmfConfig.updateConfig(commChannel)
		}
```
Then PublishOnConfigChange calls subscribeToConfigPod function which tries to connect GRPC server in an infinite loop. But this loop uses SmfConfig.Configuration.WebuiUri which can be resolved different IP addresses if NMS pod restarts.  However loop never exits and a manual intervention is required after every NMS pod restart.

func (confClient *ConfigClient) PublishOnConfigChange(mdataFlag bool) chan *protos.NetworkSliceResponse {
	confClient.MetadataRequested = mdataFlag
	commChan := make(chan *protos.NetworkSliceResponse)
	confClient.Channel = commChan
	go confClient.subscribeToConfigPod(commChan)
	return commChan
}
```

Reply:
As a result, it will restart on `commChannel := gClient.PublishOnConfigChange(false)`.

Reply:
I am completely lost. I understand high level reasoning of resolving address again but I am not able to connect to the code the way you are explaining. Give me some time.

Reply:
I guess we need the change the way Client is initialized.
Clubbing below 2 lines in single call will help in getting ...But I do not think how your change works..or why it works..

            gClient := client.ConnectToConfigServer(SmfConfig.Configuration.WebuiUri)
            commChannel := gClient.PublishOnConfigChange(false)